### PR TITLE
NEPT-2168: Delete sub jobs along with sub jobs

### PR DESCRIPTION
**Pull request by dxvargas**
### Description

Delete sub jobs along with sub jobs

### Change log

- Added: delete method in TMGMTPoetryJobController
- Added: Behat test

### CODE REVIEW

File: profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc [hunk 1]
```diff
@@ -12,6 +12,27 @@
...
+  public function delete($ids, $transaction = NULL) {
+
+    if (!is_array($ids)) {
+      $ids = [$ids];
+    }
+
+    $ids_iterate = $ids;
+    foreach ($ids_iterate as $id_iterate) {
+      $related_jobs = tmgmt_poetry_obtain_related_translation_jobs(array(), 'SUB_' . $id_iterate, TRUE);
+      foreach ($related_jobs as $related_job) {
```
Comment (drupol) [inline profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc:30]: Looping over an object is "ugly", there should be a better way to do that I guess ?

Comment (aritomelo) [reply profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc:30]: I will double check... thx @drupol!

**Comment by aritomelo** [comment]
@enriquelacoma It is ready for you to continue with CR.

**Review by enriquelacoma — DISMISSED** [review]

File: profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc (hunk 1 above)
Comment (dxvargas) [reply profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_poetry/controller/tmgmt_poetry.controller.job.inc:30]: Yes, it's better to use "->fetchAll()" like in other places this function is used.

**Review by Fefaine — APPROVED** [review]
Approving based on jira comments

**Review by ku-enza — APPROVED** [review]
Ok based on jira comment and CI test!
Ok with function delete added and behat test added.

**Pull request merged**